src/main/java/com/smartbear/plugins/swaggerhub/PublishToHubAction.java

```diff
@@ -88,40 +99,73 @@ private boolean publishApi(RestService restService) throws IOException {
                 }
             }
 
-            Swagger2Exporter exporter = new Swagger2Exporter(restService.getProject());
-            String result = exporter.exportToFileSystem(Files.createTempDir().getAbsolutePath(), versionId, "json", new RestService[]{restService}, restService.getBasePath());
+            SwaggerExporter exporter = null;
+            if (dialog.getValue(Form.OAS_VERSION).equals(SWAGGER_2_0)) {
+                exporter = new Swagger2Exporter(restService.getProject());
+            } else {
+                exporter = new OpenAPI3Exporter(restService.getProject());
+            }
+            String result = exporter.exportToFileSystem(Files.createTempDir().getAbsolutePath() + File.separator + "api-docs.json", versionId,
+                    "json", new RestService[]{restService}, restService.getBasePath());
 
             LOG.info("Created temporary Swagger definition at " + result);
-            HttpPost post = new HttpPost(uri);
-            post.setEntity(new FileEntity(new File(result, "api-docs.json"), "application/json"));
-            post.addHeader("Authorization", apikey);
+            String apiKey = "";
+            try {
+                apiKey = getApiKey(login, password);
+            } catch (Exception e) {
+                Logging.logError(e);
+                UISupport.showErrorMessage(GETTING_API_KEY_ERROR);
+                return false;
+            }
+
+            if (remember) {
+                Workspace workspace = ApplicationEnvironment.getWorkspace();
```

Comment:
Is it fine that here you use workspace settings, but for initialization you use restService.getProject().getSettings()?